lib/attrsets.nix

```diff
@@ -22,12 +20,14 @@ rec {
        => 6
   */
   attrByPath = attrPath: default: e:
-    let attr = head attrPath;
+    let
+      attr = head attrPath;
     in
-      if attrPath == [] then e
-      else if e ? ${attr}
-      then attrByPath (tail attrPath) default e.${attr}
-      else default;
+      if attrPath == [ ] then e
+      else
+        if e ? ${attr}
+        then attrByPath (tail attrPath) default e.${attr}
```

Comment:
not sure if it's worth the extra complexity to handle the `else if` case. the flat version looks nicer.

Reply:
I agree this case would be better with `else if` but not worth it. I have commented on a larger example.